src/transformers/hf_argparser.py

```diff
@@ -1,13 +1,20 @@
 import dataclasses
+import json
+import sys
 from argparse import ArgumentParser
 from enum import Enum
+from pathlib import Path
 from typing import Any, Iterable, NewType, Tuple, Union
 
 
 DataClass = NewType("DataClass", Any)
 DataClassType = NewType("DataClassType", Any)
 
 
+def trim_suffix(s: str, suffix: str):
+    return s if not s.endswith(suffix) or len(suffix) == 0 else s[: -len(suffix)]
+
```

Comment:
Is there any advantage of using this over using the buil-tin `pathlib` functionality? In pathlib with can easily remove an extension like so:

```python
p = Path('/some/path/to/file.txt')
p = p.with_suffix("")  # this means you can also easily replace the suffix, e.g. p = p.with_suffix(".args")
# /some/path/to/file
```